cmd/helmhook/hook/utils.go

```
@@ -60,6 +60,15 @@ func GetKubeBlocksDeploy(ctx context.Context, client *kubernetes.Clientset, ns s
 	return &deployments.Items[0], nil
 }
 
+// IgnoreKubeblocksHook checks if KubeBlocks deployment exists.
+func IgnoreKubeblocksHook(ctx context.Context, client *kubernetes.Clientset, ns string) bool {
+	deploy, err := GetKubeBlocksDeploy(ctx, client, ns, kubeblocksAppComponent)
+	if err != nil {
+		return false
```

The hook will continue running if an error occurs (e.g. network errors). Is it an expected behavior?

---

It is expected. If an error occurs, it should be handled by hook job. The hook job may be processed successfully, or it may fail and try again.

The only case in which upgrading can be ignored: the kubeblocks component is not deployed.

---

Ok, but if a network error occurs, you will not know whether the deployment exists. It would be better to return this error to main() and exit quickly.